dmipy/core/modeling_framework.py

```diff
@@ -863,7 +863,7 @@ def _check_acquisition_scheme_has_b0s(self, acquisition_scheme):
             raise ValueError('acquisition scheme must have b0-measurements '
                              'for signal-attenuation-based model fitting.')
 
-    def _construct_convolution_kernel(self, **kwargs):
+    def _construct_convolution_kernel(self, acquisition_scheme, **kwargs):
```

Comment:
You were right that leaving it like this breaks the code.

Please change this line to:
 `_construct_convolution_kernel(self, acquisition_scheme=None, **kwargs):`

and then right before line 895 please put:
```
if acquisition_scheme is None:
    acquisition_scheme = self.scheme
```